guidance/_optional_imports.py

```diff
@@ -0,0 +1,13 @@
+import importlib
+import typing
+
+
+def optional_import(name: str):
```

Comment:
This is a highly simplified version of the [code from pandas](https://github.com/pandas-dev/pandas/blob/576302378541230f63a0ad782ed8df399f488a00/pandas/compat/_optional.py#L91)

Reply:
And.... it's not working with `mypy` which must not be attempting to execute this when doing its type checks. Doing more reading

Reply:
I don't see you calling this anywhere, but I think you should return both the module and a Boolean indicating whether or not it was imported.

Reply:
Could you point me at that, please?

Reply:
I have now removed this file, since (as you say) it just doesn't work. I've dug through the pandas code a bit, and the equivalent there was only being called from within functions (so far as I could tell), not being used top-level (and for return types) as here.